e2e/_suites/fleet/elastic_agent_deploy.go

```diff
@@ -23,64 +23,47 @@ func (fts *FleetTestSuite) anAgentIsDeployedToFleet(image string) error {
 	if image == "debian" {
 		installerType = "deb"
 	}
-	fts.BeatsProcess = ""
 
-	// FIXME: We need to cleanup the steps to support different operating systems
-	// for now we will force the zip installer type when the agent is running on windows
-	if runtime.GOOS == "windows" && common.Provider == "remote" {
-		installerType = "zip"
-	}
-	return fts.anAgentIsDeployedToFleetWithInstallerAndFleetServer(installerType)
+	return fts.deployAgentToFleet(InstallerType(installerType))
 }
 
 func (fts *FleetTestSuite) anAgentIsDeployedToFleetOnTopOfBeat(beatsProcess string) error {
-	installerType := "tar"
-
-	// FIXME: We need to cleanup the steps to support different operating systems
-	// for now we will force the zip installer type when the agent is running on windows
-	if runtime.GOOS == "windows" && common.Provider == "remote" {
-		installerType = "zip"
-	}
-
-	fts.BeatsProcess = beatsProcess
-
-	return fts.anAgentIsDeployedToFleetWithInstallerAndFleetServer(installerType)
+	return fts.deployAgentToFleet(InstallerType("tar"), BeatsProcess(beatsProcess))
 }
 
 // supported installers: tar, rpm, deb
 func (fts *FleetTestSuite) anAgentIsDeployedToFleetWithInstaller(installerType string) error {
-	fts.BeatsProcess = ""
-
-	// FIXME: We need to cleanup the steps to support different operating systems
-	// for now we will force the zip installer type when the agent is running on windows
-	if runtime.GOOS == "windows" && common.Provider == "remote" {
-		installerType = "zip"
-	}
-
-	return fts.anAgentIsDeployedToFleetWithInstallerAndFleetServer(installerType)
+	return fts.deployAgentToFleet(InstallerType(installerType))
 }
 
 // supported installers: tar, rpm, deb
 func (fts *FleetTestSuite) anAgentIsDeployedToFleetWithInstallerAndTags(installerType string, flags string) error {
-	fts.BeatsProcess = ""
+	return fts.deployAgentToFleet(InstallerType(installerType), Flags(flags))
+}
 
-	// FIXME: We need to cleanup the steps to support different operating systems
-	// for now we will force the zip installer type when the agent is running on windows
-	if runtime.GOOS == "windows" && common.Provider == "remote" {
-		installerType = "zip"
+func (fts *FleetTestSuite) deployAgentToFleet(opts ...DeploymentOpt) error {
```

Comment:
It seems like `opts` are not used later in the function. Only logged.

Reply:
It's used here: https://github.com/elastic/e2e-testing/pull/2879/files#diff-826ef9ac59e049ffe371ffe0262603320d60f544c3263af228aaded6598b4bfdR53

it loops through the opts using range, and for each passed opt, it applies it. You can find an example for functional opts here: https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md